### PR TITLE
Menubar improvements

### PR DESCRIPTION
**Pull request by ceiphr**
This PR is intended to be a temporary fix for issue https://github.com/KryDos/todoist-linux/issues/125. Users can now enabled the menubar in the config, so those who use alt+click functionality won't be bothered by that bug anymore if they are willing to disable the menubar which offers them zoom in/out.

The menubar has also been updated to look more formalized compared to common electron menubars found in other applications.

~Finally, this PR also updates electron to v10.1.5 and electron-packager to v15.1.0. The reason being that we can't work with electron to fix upstream errors for https://github.com/KryDos/todoist-linux/issues/125 and https://github.com/KryDos/todoist-linux/issues/127 without using a current version of electron. From my testing, there are no issues resulting from these updates, but I would recommend @KryDos to set up some CI testing to cover all the bases.~

The electron upgrade introduces bugs. It will become a separate PR.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -174,10 +174,17 @@ function createWindow () {
...
-      label: 'File',
+      label: '&File',
       submenu: [
+        { role: 'undo' },
+        { role: 'redo' },
+        { type: 'separator' },
+        { role: 'cut' },
+        { role: 'copy' },
+        { role: 'paste' },
+        { type: 'separator' },
```
Comment (mikkovedru) [inline src/main.js:185]: Do I understand correctly that the typical roles like this are automatically given keyboardable labels and don't require separate definitions akin to `label: '&Undo'`?

File: src/main.js [hunk 2]
```diff
@@ -196,44 +203,37 @@ function createWindow () {
...
+        { role: 'resetzoom' },
+        { 
+          role: 'zoomin',
+          accelerator: 'commandOrcontrol+='
         },
+        { role: 'zoomout' },
+        { type: 'separator' },
+        { role: 'togglefullscreen' },
         {
           label:'Show/Hide',
```
Comment (mikkovedru) [inline src/main.js:227]: Please change `label:'Show/Hide'` to `label:'&Show/Hide'`.

File: src/main.js [hunk 3]
```diff
@@ -242,21 +242,26 @@ function createWindow () {
...
-            shell.openExternal('https://github.com/krydos/todoist-linux/releases');
+            shell.openExternal('https://github.com/KryDos/todoist-linux/issues/new');
           },
         },
         {
-          label: 'Report an issue',
+          label: 'Changelog',
           click:  function() {
-            shell.openExternal('https://github.com/KryDos/todoist-linux/issues/new');
+            shell.openExternal('https://github.com/krydos/todoist-linux/releases');
```
Comment (mikkovedru) [inline src/main.js:253]: It's interesting how in those two cases you made the exact opposite decision on how to change "krydos" (`krydos` to `KryDos`; and `KryDos` to `krydos`). :-) 

For what it's worth, my browser displays the address bar as `KryDos`.

And just so you know, I don't care if you change it or leave it be. It was just an peculiar observation. :-)

**Review by mikkovedru — COMMENTED** [review]
Just some nitpicking comments based on code. I will test the code itself a bit later this week.

**Comment by KryDos** [comment]
wow, you guys awesome. Thanks @ceiphr for the PR and thanks @mikkovedru for the review. I'll take a look during the weekend.

File: src/main.js (hunk 1 above)
Comment (ceiphr) [reply src/main.js:185]: I don't think so. They would likely need the special label.

File: src/main.js [hunk 4]
```diff
@@ -196,44 +203,37 @@ function createWindow () {
...
-        {
-          label:'Zoom Out',
-          role: 'zoomout',
-          accelerator: 'CommandOrControl+-'
-        },
-        {
-          label:'Reset Zoom',
-          role: 'resetzoom',
-          accelerator: 'CommandOrControl+0'
+          label:'Reload',
```
[the remark this answers is not in the document]
Comment (ceiphr) [reply src/main.js]: Done.

File: src/main.js (hunk 2 above)
Comment (ceiphr) [reply src/main.js]: Done.

File: src/main.js (hunk 3 above)
Comment (ceiphr) [reply src/main.js]: Nice catch. IDK how I managed to change the casing, but now they are the same.

**Comment by m-danya** [comment]
The alt button really doesn't trigger menu anymore, that's awesome! But I've faced an error: when I click on 'Preferences' in the tray button option, i get this:
image

**Comment by ceiphr** [comment]
@m-danya this may be a bug with the electron upgrade. My last final is tomorrow, I'll do some testing and update this PR after that.

Thank you for testing this build!

**Comment by m-danya** [comment]
@ceiphr Good luck!

**Comment by ceiphr** [comment]
Alright, so I think I know a good way to put this issue to rest. 

I'm going to revert the electron upgrade and make it a separate PR since it seems to be creating issues with some libraries. In the future, we can make some CI tests to ensure we catch all the bugs.

As for the menubar and zoom in/out functionality, we'll use the config setting. The menubar will be enabled by default (should we disable it by default?), and will be toggle-able.

I'll have this PR updated by tomorrow or Friday. I will see if I can make CI tests during my break.

**Comment by ceiphr** [comment]
@KryDos I've reverted the electron upgrade, so there shouldn't be any issues with this PR now. I'll make the upgrade a separate PR for when we can get CI testing set up.

@m-danya Feel free to try out the build now. Preferences work properly, but please let me know if you're still getting the blank window as I was unable to reproduce that issue.

**Comment by m-danya** [comment]
@ceiphr Hi! Thanks for the quick update, but the menubar option doesn't work for me:

image

I might have done something wrong: I've deleted ~/.config/.todoist-linux.json and repeated all the steps from https://github.com/KryDos/todoist-linux/issues/125#issuecomment-748575814, but it doesn't work..

ALSO It gives me that error and when I click on 'Ok' the screen goes white and I have to restart the application to get the login page **again**, so I don't have a possibility to get into my tasks at all :(

image

I hope you'll find the solution, thanks!

**Comment by ceiphr** [comment]
@m-danya

I fixed the bug with the setting not being recognized.

As for the login issue, I just started experiencing the same thing. [This Stack Overflow post](https://stackoverflow.com/questions/31446782/how-to-clear-the-cache-data-in-electronatom-shell) told me how to remove the cache for the electron app. I managed to find a few electron and electron development folders in `.config` and deleted them (also a Todoist folder, but maybe that was for the official snap?). That solved that issue for me. 

I'm assuming this bug was the result of running a development version of electron v10 before downgrading to v4. Especially since none of the code for this PR relates to electron storage/caching or anything session/login related.

Please let me know if that solves your login/session issue.

**Comment by m-danya** [comment]
@ceiphr Everything works now, thanks! I've deleted `Electron`, `Electron API Demos` and `Todoist` folders from `~/.config` and reinstalled the package.
I am delighted with your work, how quickly you fixed the problem every time! Thank you for your project! I'll stay at this PR and I watch for bugs, if something goes wrong, I'll tell you.

**Pull request closed without merging**